### ISSUE TITLE
[Linux] Tray issues

### ISSUE THREAD

**Issue by anatoli26** [keepassxreboot/keepassxc#104]
There's something wrong with how the minimize to tray behaves on Ubuntu 16.04 (not tried it on other OS).

TL;DR
1. Start minimized first asks to unlock the DB (IMO the app window should NOT appear if requested to start minimized). _Update_: tracked under #105.
2. Tray icon not appearing/updating
3. Unnecessary inter-dependencies for tray options/start minimized. _Update_: tracked under #64.
4. (reported in a separate issue #102) Tray menu not working

When all 4 options are checked (start minimized), the app starts with a normal window and asks for pass. Once the DB is unlocked, the app minimizes and NO tray icon appears. It appears as if the app exits/crashes, though the process is still running. I suppose that start minimized works this way: first the app starts with a normal window and asks for pass, then, after unlocking the DB, it minimizes. I believe it should start minimized and the unlock window should appear when requested (either via the new 'unlock on auto-type' functionality or via toggling the main window).

Then there's another problem: the tray icon doesn't appear even with just the first option (Show a system tray icon) checked. Consider this sequence:

The option is enabled, we start the KPXR (DB locked) -> no icon in the tray. We enter to the options, disable the option, enable the option, click [OK] to save the config -> no change. We disable the option, click on [OK] to save the config, we open the config and enable the option, click on [OK] to save the config -> the tray icon appears (grayed, meaning the DB is locked) -> we unlock the DB -> icon not changed -> we open the config, disable/save/enable/save -> the icon changes to green (meaning the DB is unlocked).

Same happens on locking the DB: the icon doesn't change, but after  disable/save/enable/save of the option, the icon is gray again.

With respect to the dependency tree of the tray options, IMO the 3rd sub-option (`minimize to tray on exit`) should be on the same level as the 2nd (`minimize to tray on minimize`) as they are independent, though related. I may want to be able to minimize the app to taskbar on minimize, but minimize to tray on close (a rather common combination).

The 4th option (`minimize on start`), IMO, shouldn't even depend on tray icon at all (though be connected). I may want to minimize the app on startup, no matter if it minimizes to tray or taskbar. If the 2nd option is selected (`minimize to tray on minimize`), then `minimize on start` should minimize to tray. If not, it should minimize to taskbar as if no tray is enabled.

Then, there is another problem already filled in a separate issue: #102.

**Comment by TheZ3ro** [comment]
Hi, next time please fill the Issue Template that will show up when you create a new issue so we can easily detect Issue.

Can you report your environment as follow: (just copy this, paste in a comment and fill the list, thanks) 
```
## Your Environment
* KeePassXR version/commit used: 
* Qt version (e.g. Qt 5.3): 
* Compiler (e.g. Clang++3.6.0): 
* Operating System and version: 
* Desktop Environment:
```

---
For the first sub-issue:
Ok It's an issue and I can reproduce it easily. I will open a new issue for that (since this one is very full and hard to understand/keep trak)

---
For the second sub-issue
On my xUbuntu16.04 it works fine. You need to fill the environment list so I can reproduce it.

---
For the third sub-issue:
There is already an Issue here -> https://github.com/keepassxreboot/keepassxc/issues/64

**Comment by anatoli26** [comment]
@TheZ3ro, great for the 1st sub-issue. With respect to the 3rd sub-issue, I have overlooked it. Anyway, IMO, it describes what reported here only partially, i.e. just the dependency of start-minimized on the hide-on-app-exit. I do agree it's better to track the issues under separate numbers, I'll add details there and update the initial post here with references.

With respect to the 2nd sub-issue, my env is as follows:

* KeePassXR version/commit used: 29f9359
* Qt version (e.g. Qt 5.3): Qt version 5.5.1, QMake version 3.0
* Compiler (e.g. Clang++3.6.0): gcc 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.4)
* Operating System and version: Ubuntu 16.04.1 LTS
* Desktop Environment: Unity 7.4.0
* Tray: libappindicator1/libappindicator3-1

**Comment by Manko10** [comment]
Can you please test my fixes from pull request #109? At least the first and third issue should be fixed now (see annotated commits in the respective issue reports). I haven't exactly addressed the second problem you are having, but please test if it still occurs. If it does, it's a deeper problem that deserves a separate issue report to analyze and keep track of it.
I'm therefore closing this. If you are still having issues with 1 and 3, please refer to issues #105 and #64. For 2, please either reopen or better create a fresh report.

**Comment by anatoli26** [comment]
Hi Janek, great work! Sub-issues 1 and 3 are resolved now, 2 and 4 need further testing. First time I started the app, the toggle and quit tray menu items worked, then I started the app again, app went directly to tray (was testing start+minimized + minimize-to-tray) and the tray icon hasn't appeared. Then turning tray icon option off and on got the tray back, but the menu never worked again and the tray icon was not updating (on unlock/lock). I'll perform more tests and open a new unifying issue for both items 2 and 4.
